requirements.txt

```diff
@@ -1,2 +1,3 @@
 requests==2.27.1
 requests-toolbelt==0.9.1
+typing_extensions; python_version<"3.8"
```

Comment:
This works in testing, but I assume we'd also need this in `setup.py` for the installed package for users.

However, let me just take another look because if we're only doing this for `Literal`, it might make more sense to add a separate argument in addition to `action`, to keep typing clearer. (Also we'd need to change the docstrings otherwise)

Reply:
Ah yes, you're right. Let me know what you decide and I will update things accordingly!

Reply:
@nejch Any news on this?

Reply:
Thanks for your patience @TCatshoek!

I just took another look at this and I feel we should maybe leave `action` a Callable only because we do this later:
https://github.com/python-gitlab/python-gitlab/blob/792cee939843d8df4c87bb8068be147ec97fabac/gitlab/utils.py#L44-L46

So I think that might be a bit confusing. How about an explicit `iterator: bool = False` argument that would follow the `streamed` logic a bit more? This would also get rid of the typing dependency we discussed as I see some people don't like having even the tiniest extra dependencies.. :)

Thanks again for working on this! Let me know what you think.

Reply:
Just had some discussion with @JohnVillalovos and we're thinking of intruducing an `iterator=True` for our `list()` calls where we currently use `as_list=False` (which is kind of unclear in what it really does currently), so this would then make it a consistent interfaces as well. So I think a bool with `iterator` is the way to go,